### PR TITLE
#160577586 Implement search functionality

### PR DESCRIPTION
**Pull request by imireallan**
## What does this PR do?
This PR adds the search functionality which enable users to query articles

### Why this is important
Looking for article(s) in their dozens can be daunting. Would it be easier if we had a search functionality? I bet it would.

### As a User
I want to be able to search for articles by author, article title and tag/keyword or search by filtering.

#### Scenario:
Any user (authenticated or not) should be able to search for articles on our platform .

## How should this be manually tested?

1. Log in to the app.
2. On the NavBar, click the search icon and start typing

## What are the relevant pivotal tracker stories?

The relevant pivotal story is [here](https://www.pivotaltracker.com/n/projects/2198535/stories/160577586)

### CODE REVIEW

File: src/containers/ArticleListing/index.js [hunk 1]
```diff
@@ -21,13 +21,26 @@ export class ArticleListing extends React.Component {
...
       url, fetchArticles, params, listName,
     } = this.props;
     fetchArticles(url, params, listName);
   };
 
+  load() {
+    const {
+      fetchArticles, url, params, listName,
+    } = this.props;
+    fetchArticles(url, params, listName);
```
Comment (paulrimiru) [inline src/containers/ArticleListing/index.js]: This line has not been covered yet

File: src/pages/Search/search.test.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import mockStore from '../../utils/redux_mock_store';
+
+const store = mockStore();
+
+const props = {
+  fetchArticles: jest.fn(),
+
+};
+
+
```
Comment (paulrimiru) [inline src/pages/Search/search.test.js]: Remove extra line

File: src/pages/Search/search.test.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import { Provider } from 'react-redux';
+
+import { Search } from './index';
+import mockStore from '../../utils/redux_mock_store';
+
+const store = mockStore();
+
+const props = {
+  fetchArticles: jest.fn(),
+
```
Comment (paulrimiru) [inline src/pages/Search/search.test.js]: Remove extra line

File: src/pages/Search/search.test.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+};
+
+
+describe('<Search />) container', () => {
```
Comment (paulrimiru) [inline src/pages/Search/search.test.js]: Test cases should read as a sentence replace this with `The Search container`

**Review by paulrimiru — CHANGES_REQUESTED** [review]
Make the changes, above also make sure that you refactor the test cases to read as a sentense

File: src/pages/Search/search.test.js (hunk 2 above)
Comment (imireallan) [reply src/pages/Search/search.test.js]: on it

File: src/pages/Search/search.test.js (hunk 3 above)
Comment (imireallan) [reply src/pages/Search/search.test.js]: on it

File: src/pages/Search/search.test.js (hunk 4 above)
Comment (imireallan) [reply src/pages/Search/search.test.js]: done

File: src/containers/ArticleListing/index.js (hunk 1 above)
Comment (imireallan) [reply src/containers/ArticleListing/index.js]: The load function is not used anywhere, I forgot to remove it..so am removing it now...

File: src/pages/Search/search.test.js (hunk 2 above)
Comment (imireallan) [reply src/pages/Search/search.test.js]: done

File: src/pages/Search/search.test.js (hunk 3 above)
Comment (imireallan) [reply src/pages/Search/search.test.js]: done

**Review by paulrimiru — APPROVED** [review]

**Pull request merged**